lib/src/builder/parsing/members/factory.dart

```diff
@@ -45,6 +45,10 @@ class BoilerplateFactory extends BoilerplateMember {
 
   bool get hasFactoryAnnotation => node.hasAnnotationWithName('Factory');
 
+  bool get isFunctionComponentFactory =>
+      node.variables.firstInitializer is MethodInvocation &&
+      (node.variables.firstInitializer as MethodInvocation).methodName.name == 'uiFunction';
```

Comment:
FYI `tryCast` can be used to avoid extra casting these checks a little more compact
```suggestion
      node.variables.firstInitializer?.tryCast<MethodInvocation>()?.methodName?.name == 'uiFunction';
```